lint/package.go

```diff
@@ -182,17 +183,41 @@ func (p *Package) scanSortable() {
 	}
 }
 
-func (p *Package) lint(rules []Rule, config Config, failures chan Failure) {
+func (p *Package) lint(rules []Rule, config Config, failures chan Failure) error {
 	p.scanSortable()
 	var wg sync.WaitGroup
+	errCh := make(chan error, 1) // Buffered channel to signal the first error
+    _, cancel := context.WithCancel(context.Background())
+
+    defer cancel() // Ensure the context is canceled at the end
+
 	for _, file := range p.files {
 		wg.Add(1)
 		go (func(file *File) {
-			file.lint(rules, config, failures)
-			wg.Done()
+			defer wg.Done()
+			if err := file.lint(rules, config, failures); err != nil {
+                select {
```

Comment:
All this code block seems to be a c&p from an AI. Careful must be taken when integrating AI generated code, it usually has subtle bugs (in particular with concurrent code)

Reply:
ok reverted that code, any idea how to handle error and stop processing other rules? If we have error in configure in any rule we should stop processing ?

Reply:
> how to handle error and stop processing other rules?

Just exit the program at the first error

> If we have error in configure in any rule we should stop processing ?

Currently, `revive` stops at the first error it encounters, I propose to keep that behavior.